### PR TITLE
Rotary testing

### PR DESCRIPTION
**Pull request by yuanandonly**
## What does this PR do?
Adds several unit tests on uncovered methods in rotary.py

## Before submitting

- [x] Did you have fun?
  - Make sure you had fun coding 🙃
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/xformers/blob/master/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [x] N/A
- [ ] Did you make sure to update the docs?
  - [x] N/A
- [x] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/xformers/blob/master/CHANGELOG.md)? (if needed)
  - [x] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: tests/test_rotary_embeddings.py [hunk 1]
```diff
@@ -21,6 +25,32 @@
 EMB = 32
 
 
+def test_helper_methods():
+    # rotate_half
+    tens = torch.tensor([[0, 1, 2, 3], [3, 1, 2, 0], [0, 1, 0, 1], [1, 0, 1, 0]])
```
Comment (dianaml0) [inline tests/test_rotary_embeddings.py:30]: nit: may be nice to format this for readability like [here](https://github.com/facebookresearch/xformers/blob/37cf11f9c8124d439b89d7bb133b0157f11ecad3/tests/test_attention_patterns.py#L232)

**Comment by blefaudeux** [comment]
hmm, I'm not sure why codecov has not reported here ? else looks good @yuanandonly, thank you !

**Comment by dianaml0** [comment]
@blefaudeux not sure either why codecov isn't reporting here, but looks like there was an increase of .15% from the previous commit

Screen Shot 2022-06-03 at 4 31 33 PM
Screen Shot 2022-06-03 at 4 32 07 PM
:

**Review by dianaml0 — APPROVED** [review]
This looks good, thanks @yuanandonly!

File: tests/test_rotary_embeddings.py (hunk 1 above)
Comment (yuanandonly) [reply tests/test_rotary_embeddings.py:30]: The pre-commit formatting configurations (black specifically) keep reverting my formatting changes back to what is shown above. @blefaudeux should we leave it as is?

**Pull request merged**